var/spack/repos/builtin/packages/py-pennylane-lightning-kokkos/package.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack.package import *
+
+
+class PyPennylaneLightningKokkos(CMakePackage, PythonExtension, CudaPackage, ROCmPackage):
+    """The PennyLane-Lightning-Kokkos plugin provides
+    a fast state-vector simulator with Kokkos kernels."""
+
+    homepage = "https://docs.pennylane.ai/projects/lightning-kokkos"
+    git = "https://github.com/PennyLaneAI/pennylane-lightning-kokkos.git"
+    url = "https://github.com/PennyLaneAI/pennylane-lightning-kokkos/archive/refs/tags/v0.29.1.tar.gz"
+
+    maintainers("vincentmr")
+
+    version("main", branch="main")
+    version("0.29.1", sha256="f51ba7718defc7bb5064f690f381e04b2ec58cb09f22a171ae5f410860716e30")
+
+    # kokkos backends
+    backends = {
+        "cuda": [False, "Whether to build CUDA backend"],
+        "openmp": [False, "Whether to build OpenMP backend"],
+        "openmptarget": [False, "Whether to build the OpenMPTarget backend"],
+        "pthread": [False, "Whether to build Pthread backend"],
+        "rocm": [False, "Whether to build HIP backend"],
+        "serial": [True, "Whether to build serial backend"],
+        "sycl": [False, "Whether to build the SYCL backend"],
+    }
+
+    for backend in backends:
+        deflt_bool, descr = backends[backend]
+        variant(backend.lower(), default=deflt_bool, description=descr)
+        depends_on(f"kokkos+{backend.lower()}", when=f"+{backend.lower()}", type=("run", "build"))
+
+    # CUDA
+    for val in CudaPackage.cuda_arch_values:
+        depends_on("kokkos cuda_arch={0}".format(val), when="cuda_arch={0}".format(val))
+    # Use +wrapper when not %clang %cce
+    depends_on("kokkos+wrapper", when="%gcc+cuda")
+
+    # ROCm
+    for val in ROCmPackage.amdgpu_targets:
+        depends_on("kokkos amdgpu_target={0}".format(val), when="amdgpu_target={0}".format(val))
+
+    conflicts(
+        "+cuda",
+        when="+rocm",
+        msg="CUDA and ROCm are not compatible in PennyLane-Lightning-Kokkos.",
+    )
+
+    # build options
+    extends("python")
+    variant(
+        "build_type",
+        default="Release",
+        description="CMake build type",
+        values=("Debug", "Release", "RelWithDebInfo", "MinSizeRel"),
+    )
+    variant("cpptests", default=False, description="Build CPP tests")
+    variant("native", default=False, description="Build natively for given hardware")
+    variant("sanitize", default=False, description="Build with address sanitization")
+
+    # hard dependencies
+    depends_on("cmake@3.21:3.24,3.25.2:", type="build")
```

Comment:
According to CMakeLists.txt this should be:
```suggestion
    depends_on("cmake@3.20:", type="build")
```
but if you know better than that file that's fine too.

Reply:
I think you're right.